platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinScriptEvaluator.kt

```diff
@@ -398,8 +402,8 @@ class CompileKotlinScript(
         identityFileInputs: MutableMap<String, CurrentFileCollectionFingerprint>
     ): UnitOfWork.Identity {
         val identityHash = newHasher().let { hasher ->
-            IDENTITY_HASH__PROPERTIES.forEach {
-                requireNotNull(identityInputs[it]).appendToHasher(hasher)
+            identityInputs.values.forEach {
```

Comment:
👍 TIL

Reply:
I'm planning to make this the default behavior, so you don't have to care about it unless you need something special (like in `NonNormalizedIdentityImmutableTransformExecution`).

Reply:
See https://github.com/gradle/gradle/issues/27142